ansible_builder/main.py

```diff
@@ -1,19 +1,29 @@
 import os
 import yaml
+import subprocess
 import sys
 from shutil import copy
 
-
-default_base_image = 'shanemcd/ansible-runner'
+from . import constants
 
 
 class AnsibleBuilder:
-    def __init__(self, filename='execution-environment.yml', base_image=default_base_image, build_context=None):
+    def __init__(self,
+                 filename=constants.default_file,
+                 base_image=constants.default_base_image,
+                 build_context=constants.default_build_context,
+                 tag=constants.default_tag,
+                 container_runtime=constants.default_container_runtime):
         self.definition = Definition(filename=filename)
         self.base_image = base_image
-        self.build_context = build_context or os.path.join(os.getcwd(), 'context')
+        self.tag = tag
+        self.build_context = build_context
+        self.container_runtime = container_runtime
         self.containerfile = Containerfile(
-            filename='Containerfile',
+            filename={
+                'podman': 'Containerfile',
+                'docker': 'Dockerfile'
+            }[self.container_runtime],
```

Comment:
This is going to throw a key error if passed anything other than the exact strings "podman" or "docker". If we want to go this route, let's do some validation to make sure this is one or the other. Though it _may_ be desirable to have this be an arbitrary string if you have a wrapper script or something else that you use to invoke these tools, maybe some other use cases I am not thinking of.

Reply:
Yes, the argparse definition of the option should have these two choices. We can put the mapping inside the constants, and also define the options by making a list of the keys of this dict.